### PR TITLE
Add SCIM Integration Resource

### PR DESCRIPTION
**Pull request by ChrisIsidora**
Added SCIM Integration Resource (#549)

## Test Plan
* [x] acceptance tests
* [x] unit tests…

## References
See https://docs.snowflake.com/en/user-guide/scim.html

*

### CODE REVIEW

**Comment by ChrisIsidora** [comment]
@alldoami This is related to the feature request #549

File: pkg/resources/scim_integration.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+			"OKTA", "AZURE", "CUSTOM",
+		}, true),
+		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+			normalize := func(s string) string {
+				return strings.ToUpper(strings.Replace(s, "-", "", -1))
+			}
+			return normalize(old) == normalize(new)
+		},
+	},
+	"run_as_role": {
```
Comment (alldoami) [inline pkg/resources/scim_integration.go:35]: Wondering if there is a clearer name for this... possibly `provisioner_role`?

Comment (ChrisIsidora) [reply pkg/resources/scim_integration.go:35]: I wanted to keep it the same as in Snowflake itself. But good point provisioner_role is more clear

Comment (ChrisIsidora) [reply pkg/resources/scim_integration.go]: @alldoami Renamed to `provisioner_role`

**Comment by ChrisIsidora** [comment]
@alldoami can you trigger the test after the latest changes?

**Comment by ChrisIsidora** [comment]
@alldoami  Removed Enabled property which cannot be set for SECURITY INTEGRATION with type=SCIM

**Comment by ChrisIsidora** [comment]
@alldoami Added the dependent resources to the acceptance tests. Can you check if it's ok now?

**Comment by alldoami** [comment]
This would be best if you could test locally and request my review when ready.

**Comment by ChrisIsidora** [comment]
@alldoami You are right, sorry for the inconvenience. I tested the acceptance test locally, made some fixes and it seems to work now.

image

**Comment by ChrisIsidora** [comment]
@alldoami Not sure why the acc test are failing. It does work on my snowflake account. See below, I made some changes to the resource id maybe that might help. For the unit test I saw that for the Generic alter builder there was no sorting which means that my alter unit test might fail sometimes because of the random order of the statement. That should be fixed now.

image

**Comment by ChrisIsidora** [comment]
@alldoami  Seems that okta provider already exists on the acc snowflake account. I removed Okta tests

**Comment by ChrisIsidora** [comment]
@alldoami can this be merged so that I can work / update #557? Thanks in advance for your patience

File: pkg/snowflake/generic.go [hunk 2]
```diff
@@ -108,24 +109,53 @@ func (ab *AlterPropertiesBuilder) Statement() string {
...
-		sb.WriteString(fmt.Sprintf(" %s='%s'", strings.ToUpper(k), EscapeString(v)))
+	sortedStringProperties := make([]string, 0)
+	for k := range ab.stringProperties {
+		sortedStringProperties = append(sortedStringProperties, k)
 	}
+	sort.Strings(sortedStringProperties)
 
-	for k, v := range ab.stringListProperties {
-		sb.WriteString(fmt.Sprintf(" %s=%s", strings.ToUpper(k), formatStringList(v)))
+	for _, k := range sortedStringProperties {
```
Comment (jacobrheath) [inline pkg/snowflake/generic.go:118]: This is a little verbose for all the fields. Can you make a helper function that returns a sorted list of string?
```suggestion
	for _, k := range sortStrings(ab.stringProperties) {
```

`sortStrings` would look something like `func sortStrings(strs []string) []string` and would return a list of a sorted strings from the list passed in.

File: pkg/snowflake/generic.go [hunk 3]
```diff
@@ -108,24 +109,53 @@ func (ab *AlterPropertiesBuilder) Statement() string {
 
 	sb.WriteString(ab.rawStatement)
 
-	for k, v := range ab.stringProperties {
-		sb.WriteString(fmt.Sprintf(" %s='%s'", strings.ToUpper(k), EscapeString(v)))
+	sortedStringProperties := make([]string, 0)
```
Comment (jacobrheath) [inline pkg/snowflake/generic.go:112]:
```suggestion
	sortedStringProperties := []string{}
```

Comment (jacobrheath) [reply pkg/snowflake/generic.go:112]: I think this is the convention we have going.

**Review by jacobrheath — CHANGES_REQUESTED** [review]

File: pkg/snowflake/generic.go (hunk 2 above)
Comment (ChrisIsidora) [reply pkg/snowflake/generic.go:118]: @jacobrheath I copied this from the Create function which is also in Generic, but I will try your suggestion

**Comment by ChrisIsidora** [comment]
@alldoami  and @jacobrheath Can you please review the changes requested. Thanks!

**Review by alldoami — APPROVED** [review]

**Pull request merged**